hack/dockerfiles/lint.Dockerfile

```diff
@@ -3,10 +3,8 @@
 ARG GO_VERSION=1.17
 
 FROM golang:${GO_VERSION}-alpine
-RUN apk add --no-cache gcc musl-dev yamllint
+RUN apk add --no-cache gcc musl-dev
 RUN wget -O- -nv https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.36.0
 WORKDIR /go/src/github.com/docker/buildx
 RUN --mount=target=/go/src/github.com/docker/buildx --mount=target=/root/.cache,type=cache \
   golangci-lint run
-RUN --mount=target=/go/src/github.com/docker/buildx --mount=target=/root/.cache,type=cache \
-  yamllint -c .yamllint.yml --strict .
```

Comment:
looks like yamllint is not happy even if it's valid: https://github.com/docker/buildx/runs/5785153351?check_suite_focus=true#step:3:135

```
#14 [stage-0 6/6] RUN --mount=target=/go/src/github.com/docker/buildx --mount=target=/root/.cache,type=cache   yamllint -c .yamllint.yml --strict .
#14 0.744 ./.github/ISSUE_TEMPLATE/bug.yml
#14 0.744   95:1      error    trailing spaces  (trailing-spaces)
#14 0.744   116:1     error    trailing spaces  (trailing-spaces)
#14 0.744 
#14 ERROR: executor failed running [/bin/sh -c yamllint -c .yamllint.yml --strict .]: exit code: 1
```

I think we can remove this linter. It that was added in https://github.com/docker/buildx/pull/451 for GitHub Actions workflow but know GitHub reports YAML syntax issues. Maybe I could instead add an exception instead for the issue template folder. Let me know if it sgty.

Reply:
Curious if there were actually trailing spaces though, or was it because of the empty "code" block?

Reply:
Yes that's because the empty code block :disappointed: